pkg/apicheck/check.go

```diff
@@ -129,12 +130,17 @@ func (c *ApiConnectivityCheck) getWorkerPeersResponse() peers.Response {
 
 	c.config.Log.Info("Error count exceeds threshold, trying to ask other nodes if I'm healthy")
 	peersToAsk := c.config.Peers.GetPeersAddresses(peers.Worker)
-	if peersToAsk == nil || len(peersToAsk) == 0 {
-		c.config.Log.Info("Peers list is empty and / or couldn't be retrieved from server, nothing we can do, so consider the node being healthy")
-		// TODO: maybe we need to check if this happens too much and reboot
+	if peersToAsk == nil && c.config.MinPeersForRemediation != 0 || len(peersToAsk) < c.config.MinPeersForRemediation {
```

Comment:
For me it's a bit tricky to read.
Hopefully I didn't miss anything (so feel free to correct me if I did :sweat_smile: )

I suggest something along these lines:
```
if  len(peersToAsk) == 0 {
   //if MinPeersForRemediation == 0 ...
   if c.config.MinPeersForRemediation == 0{
     return peers.Response{IsHealthy: false, Reason: peers.UnHealthyBecauseNodeIsIsolated}
   } 
  //TODO ...
   c.config.Log.Info("Peers list is empty and / or less than the minimum required peers for remediation, so consider the node being healthy")
 }
    
```

I think that this way it's clear we are addressing 2 different use cases where no peers where found (i.e peersToAsk is empty)

Reply:
I broke it up into the two cases, and removed checking for nil, and added some more comments to hopefully make it more understandable.

I didn't do the original code, but definitely can see where you are coming from.  Hope this makes it more clear.